## Table of Contents
- [Introduction](#introduction)
- [Data Wrangling](#wrangling)
    - [Gather](#gather)
    - [Assess](#assess)
    - [Clean](#clean)
    - [Analyze](#analyze)
    - [Visualize](#visualize)
- [Conclusions](#conclusions)

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sb

## I) Introduction <a id = "introduction">

**Aim:** Analyze absolute difference and (possibly) margin of error between stock market forecast of price returns and actual stock market price returns.

I will be analyzing quarterly price returns within the past 20 years for the firms present in the S&P 500 2019 Index.

> At first, I wanted to analyze the forecasted vs. actual price earnings of the S&P in its entirety for the past 20 years. However, considering that firms continuously enter and leave stock indices every year, there would be varying levels of inconsistencies and marginal errors when comparing annual S&P returns alone. To combat this problem, I have isolated these two approaches:
- Analyze the historical earnings of *only* the firms present in the S&P 2019 Index
- Keep track of all firms that were present in the S&P for the past 20 years. Keep track of how many times each firm appeared in the Index and for those with the least count, analyze them individually on how they differ from the firms that stayed for longer.


## II) Data Wrangling <a id="wrangling"></a>

To gather the data depicted under the `./data` folder, I used Bloomberg Excel functions.

### A) Gather <a id = "gather"></a>
> **APPROACH 1:** Focus on the firms that appear in the 2019 S&P Index and analyze their forecasted vs. actual price earnings for the last 20 years.

To ensure consistency in analysis among multiple firms, I divide both the forecasted and actual price earning dates by *calendar period* instead of fiscal period. This is because fiscal period differs by firm whereas calendar period is consistent by dates. 

### Forecasted Stock Returns (201? - 201?)

### Historic Stock Returns

In [16]:
sp_2019_path = './data/'
df_sp_2019 = pd.read_csv(sp_2019_path + 'sp-2019.csv')

## B) Assess

### Forecasted Stock Returns

### Historic Stock Returns

In [13]:
#generate 10 random samples 
df_sp_2019.sample(10)

date  A UN Equity  AAL UW Equity  AAP UN Equity  AAPL UW Equity  \
30   9/29/2006      22.0256            NaN         32.940         11.0043   
1    6/30/1999          NaN            NaN            NaN          1.6540   
45   6/30/2010      20.3299            NaN         50.180         35.9329   
76   3/29/2018      66.9000          51.96        118.550        167.7800   
49   6/30/2011      36.5481            NaN         58.490         47.9529   
22   9/30/2004      14.5332            NaN         22.933          2.7679   
79  12/31/2018      67.4600          32.11        157.460        157.7400   
42   9/30/2009      19.9009            NaN         39.280         26.4814   
73   6/30/2017      59.3100          50.32        116.590        144.0200   
19  12/31/2003      19.7010            NaN         27.133          1.5264   

    ABBV UN Equity  ABC UN Equity  ABMD UW Equity  ABT UN Equity  \
30             NaN        21.9296          14.790        23.2347   
1              NaN         6.1859           6.875        20.3630   
45             NaN        31.7500           9.680        22.3830   
76           94.65        86.2100         290.990        59.9200   
49             NaN        41.4000          16.200        25.1773   
22             NaN        13.0292           8.850        20.2681   
79           92.19        74.4000         325.040        72.3300   
42             NaN        22.3800           9.710        23.6701   
73           72.51        94.5300         143.300        48.6100   
19             NaN        13.6211           6.910        20.8553   

    ACN UN Equity  ...  XEL UW Equity  XLNX UW Equity  XOM UN Equity  \
30          31.71  ...            NaN          21.950        67.1000   
1             NaN  ...            NaN          28.625        38.5625   
45          38.65  ...            NaN          25.260        57.0700   
76         153.50  ...          45.48          72.240        74.6100   
49          60.42  ...            NaN          36.470        81.3800   
22          27.05  ...            NaN          27.000        48.3300   
79         141.01  ...          49.27          85.170        68.1900   
42          37.27  ...            NaN          23.420        68.6100   
73         123.68  ...            NaN          64.320        80.7300   
19          26.32  ...            NaN          38.740        41.0000   

    XRAY UW Equity  XRX UN Equity  XYL UN Equity  YUM UN Equity  \
30          30.110        40.9943            NaN        18.7134   
1            9.625       155.6057            NaN         9.7297   
45          29.910        21.1821            NaN        28.0719   
76          50.310        28.7800          76.92        85.1300   
49          38.080        27.4261            NaN        39.7206   
22          25.970        37.0951            NaN        14.6184   
79          37.210        19.7600          66.72        91.9200   
42          34.540        20.3918            NaN        24.2753   
73          64.840        28.7300          55.43        73.7600   
19          22.585        36.3574            NaN        12.3677   

    ZBH UN Equity  ZION UW Equity  ZTS UN Equity  
30          67.50           79.81            NaN  
1             NaN           63.50            NaN  
45          54.05           21.57            NaN  
76         109.04           52.73          83.51  
49          63.20           24.01            NaN  
22          79.04           61.04            NaN  
79         103.72           40.74          85.54  
42          53.45           17.97            NaN  
73         128.40           43.91          62.38  
19          70.40           61.33            NaN  

[10 rows x 506 columns]

In [15]:
df_sp_2019.shape

(84, 506)

**Observation:** There are 506 firms encompassing 84 




### Quality

**Missing Data**
- 


### Tidiness

## C) Cleaning

# III) Store Data

# IV) Explore Data

## Univariate

## Bivariate

## Multivariate

# V) Visualize Data